File: src/main/java/work/lclpnet/ap2/base/ApConstants.java
package work.lclpnet.ap2.base;

import net.fabricmc.loader.api.FabricLoader;

public final class ApConstants {

    public static final boolean DEVELOPMENT = FabricLoader.getInstance().isDevelopmentEnvironment();
    public static final String ID = "ap2";
    public static final String MAP_VERSION = "";

    // people
    public static final String PERSON_LCLP = "@person.lclp";
    public static final String PERSON_BOPS = "@person.bops";

    // other
    public static final String SEPARATOR = "===================================";
    public static final String SCOREBOARD_SEPARATOR = "==============";

    private ApConstants() {}
}

File: src/lib/src/main/kotlin/work/lclpnet/ap2/ApConstants.kt
package work.lclpnet.ap2

import net.fabricmc.loader.api.FabricLoader
import net.minecraft.resources.Identifier
import org.slf4j.Logger
import org.slf4j.LoggerFactory

object ApConstants {
    val DEVELOPMENT: Boolean =
        FabricLoader.getInstance().isDevelopmentEnvironment || "true" == System.getenv("AP2_DEV")
    @JvmField
    val DEBUG: Boolean = FabricLoader.getInstance().isDevelopmentEnvironment

    const val ID = "ap2"
    const val LIB_ID = "ap2-lib"
    @JvmField
    val RUNTIME_CONFIG_ID = if (DEVELOPMENT) "$ID-dev" else ID

    @JvmField
    val logger: Logger = LoggerFactory.getLogger(ID)

    // people
    const val PERSON_LCLP = "@person.lclp"
    const val PERSON_BOPS = "@person.bops"
    const val PERSON_QUADRUBO = "@person.quadrubo"

    // other
    const val SEPARATOR = "============================================="
    const val SCOREBOARD_SEPARATOR = "=============="
    const val SCOREBOARD_SEPARATOR_SM = "----------------"
    const val TABLIST_SEPARATOR = "========================"
    const val TABLIST_SEPARATOR_SM = "------------------------"

    @JvmStatic
    fun identifier(path: String): Identifier =
        Identifier.fromNamespaceAndPath(ID, path)
}
